src/core/algorithms/md/hymd/preprocessing/similarity_measure/distance_similarity_measure.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <vector>
+#include <functional>
+#include <algorithm>
+#include <unordered_set>
+#include <sstream>
+#include <cstddef>
+#include <numeric>
+
+#include "algorithms/md/hymd/preprocessing/similarity_measure/distance_similarity_measure.h"
+#include "config/exceptions.h"
+#include "model/types/date_type.h"
+
+double NumberDifference(const std::byte* left, const std::byte* right) {
+    auto left_val = model::Type::GetValue<model::Double>(left);  
+    auto right_val = model::Type::GetValue<model::Double>(right);
+
+    return std::abs(static_cast<double>(left_val - right_val));
+}
+
+double DateDifference(const std::byte* left, const std::byte* right) {
+    auto left_date = model::Type::GetValue<model::Date>(left);
+    auto right_date = model::Type::GetValue<model::Date>(right);
+
+    model::DateType date_type;
+    return static_cast<double>(abs(date_type.SubDate(left, right).days()));
+}
+
+
+namespace algos::hymd::preprocessing::similarity_measure {
+
+using SimInfo = std::map<Similarity, indexes::RecSet>;
+
+indexes::ColumnMatchSimilarityInfo DistanceSimilarityMeasure::MakeIndexes(
+        std::shared_ptr<DataInfo const> data_info_left,
+        std::shared_ptr<DataInfo const> data_info_right,
+        std::vector<indexes::PliCluster> const* clusters_right, model::md::DecisionBoundary min_sim, bool is_null_equal_null) const {
+    std::vector<model::md::DecisionBoundary> decision_bounds;
+    indexes::SimilarityMatrix similarity_matrix;
+    indexes::SimilarityIndex similarity_index;
+    auto const& data_left_size = data_info_left->GetElementNumber();
+    auto const& data_right_size = data_info_right->GetElementNumber();
+    Similarity lowest = 1.0;
+    for (ValueIdentifier value_id_left = 0; value_id_left < data_left_size; ++value_id_left) {
+        std::vector<std::pair<Similarity, RecordIdentifier>> sim_rec_id_vec;
+        std::byte const* left_value = data_info_left->GetAt(value_id_left);
+        Similarity max_distance = 0;
```

Comment:
`Similarity` is a `double` in the [0.0, 1.0] range.
```suggestion
        double max_distance = 0;
```